components/api/src/main/java/com/hotels/styx/api/extension/service/HealthCheckConfig.java

```diff
@@ -63,10 +61,18 @@ private HealthCheckConfig(Optional<String> uri,
         this.healthyThreshold = healthyThreshold.orElse(DEFAULT_HEALTHY_THRESHOLD_VALUE);
         this.unhealthyThreshold = unhealthyThreshold.orElse(DEFAULT_UNHEALTHY_THRESHOLD_VALUE);
 
-        checkArgument(this.intervalMillis >= 1, format("intervalMillis [%s] cannot be < 1 ms", intervalMillis));
-        checkArgument(this.timeoutMillis >= 1, format("timeoutMillis [%s] cannot be < 1 ms", intervalMillis));
-        checkArgument(this.healthyThreshold >= 1, format("healthyThreshold [%s] cannot be < 1", healthyThreshold));
-        checkArgument(this.unhealthyThreshold >= 1, format("unhealthyThreshold [%s] cannot be < 1", unhealthyThreshold));
+        if (this.intervalMillis < 1) {
+            throw new IllegalArgumentException(format("intervalMillis [%s] cannot be < 1 ms", intervalMillis));
+        }
+        if (this.timeoutMillis < 1) {
+            throw new IllegalArgumentException(format("timeoutMillis [%s] cannot be < 1 ms", intervalMillis));
```

Comment:
This should be `timeoutMillis` above. But I can see this mistake is also in the original code.

Reply:
Thanks, I missed that.